src/entities/SurveyTopic/decoder.ts

```diff
@@ -1,76 +1,51 @@
-      if (parsedTopic) survey.push(parsedTopic)
-      buffer = buffer.substring(separatorPos + 1, buffer.length)
-    }
-
-    return survey
-  }
-
-  private parseTopicFeedback(topicFeedback: string): TopicFeedback | null {
-    const topic = this.parseTopic(topicFeedback)
-    const parsedReaction = this.parseReaction(topicFeedback)
-
-    if (!topic || !parsedReaction) {
-      return null
-    }
-
-    return {
-      topic,
-      reaction: parsedReaction,
-    }
-  }
-
-  private parseReaction(topicFeedback: string): ReactionType | null {
-    const reaction =
-      topicFeedback.substring(topicFeedback.indexOf(TOPIC_REACTION_CONCAT) + 1, topicFeedback.length) || ''
-    return SurveyDecoder.getReaction(reaction)
-  }
+function isValidTopicFeedback(rawTopicFeedback: never) {
+  return (
+    typeof rawTopicFeedback === 'object' &&
+    rawTopicFeedback !== null &&
+    has('topic', rawTopicFeedback) &&
+    has('reaction', rawTopicFeedback)
+  )
+}
 
-  private parseTopic(topicFeedback: string): Pick<SurveyTopicAttributes, 'topic_id' | 'label'> | undefined {
-    const topicId = topicFeedback.substring(0, topicFeedback.indexOf(TOPIC_REACTION_CONCAT))
-    const topic = this.topics.find((topic) => topic.topic_id === topicId)
-    if (!topic) console.log(`Unable to parse topic feedback ${topicFeedback}`) // TODO: report error
-    return topic
-  }
+function isValidTopic(rawTopic: unknown) {
+  return typeof rawTopic === 'object' && rawTopic !== null && has('topic_id', rawTopic)
+}
 
-  private static getReaction(reaction: string): ReactionType | null {
-    const foundReaction = REACTIONS_VIEW.find((reactionView) => reactionView.icon === reaction)?.reaction || null
-    if (foundReaction && SurveyDecoder.isReactionType(foundReaction)) return foundReaction as ReactionType
-    else {
-      console.log(`Unable to parse reaction ${reaction}`) // TODO: report error
-      return null
-    }
+function isReactionType(value: string | null | undefined): boolean {
+  switch (value) {
+    case ReactionType.EMPTY:
+    case ReactionType.LOVE:
+    case ReactionType.LIKE:
+    case ReactionType.CONCERNED:
+    case ReactionType.NEUTRAL:
+      return true
+    default:
+      return false
   }
+}
 
-  private static isReactionType(value: string | null | undefined): boolean {
-    switch (value) {
-      case ReactionType.EMPTY:
-      case ReactionType.LOVE:
-      case ReactionType.LIKE:
-      case ReactionType.NEUTRAL:
-      case ReactionType.CONCERNED:
-        return true
-      default:
-        return false
+export function decodeSurvey(encodedSurvey?: Record<string, unknown>): Survey {
+  const decodedSurvey: Survey = []
+  if (!encodedSurvey || Object.keys(encodedSurvey).length === 0) return decodedSurvey
+  try {
+    const rawSurvey: unknown = encodedSurvey?.survey
+    if (Object.prototype.toString.call(rawSurvey) === '[object Array]') {
+      const arraySurvey = rawSurvey as []
+      arraySurvey.forEach((rawTopicFeedback) => {
```

Comment:
Can't you type rawTopicFeedback here to avoid reassigning?`arraySurvey.forEach((rawTopicFeedback: TopicFeedback) => {`

Reply:
I don't think that would work if it's an invalid topic feedback (or it has a weird data structure), that's why I'm casting only after the validation